src/BasketballGames/GamesRegister.java

```diff
@@ -0,0 +1,91 @@
+package BasketballGames;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+public class GamesRegister {
+    private static Game [] games = new Game[0];
+
+    public void addGame(Game game){
+        games = addGameToArray(games,game);
+    }
+
+    public Game[] getGames(){
+        return games;
+    }
+
+    public Game[] getPastGames(){
+        Game [] pastGames = new Game[0];
+        for (int i = 0; i < games.length; i++){
+            if (games[i].getDateTime().isBefore(LocalDateTime.now())){
+                pastGames = addGameToArray(pastGames, games[i]);
+            }
+        }
+        return pastGames;
+    }
+
+    public Game[] getFutureGames(){
+        Game [] futureGames = new Game[0];
+        for (int i = 0; i < games.length; i++){
+            if (games[i].getDateTime().isAfter(LocalDateTime.now())){
+                futureGames = addGameToArray(futureGames, games[i]);
+            }
+        }
+        return futureGames;
+    }
+
+    public Game[] getTodayGames(){
+        DateTimeFormatter dayFormatter = DateTimeFormatter.ofPattern("YYYY-MM-d");
+        return getGamesUsingFormatter(dayFormatter);
+    }
+
+    public Game[] getGamesOfThisWeek(){
+        DateTimeFormatter weekFormatter = DateTimeFormatter.ofPattern("W");
+        return getGamesUsingFormatter(weekFormatter);
+    }
+
+    private Game [] getGamesUsingFormatter(DateTimeFormatter formatter) {
+        Game[] gamesArray = new Game[0];
+        for (int i = 0; i < games.length; i++) {
+            if (games[i].getDateTime().format(formatter).equals(LocalDateTime.now().format(formatter))) {
+                gamesArray = addGameToArray(gamesArray, games[i]);
+            }
+        }
+        return gamesArray;
+    }
+
+    public Game[] getGamesAtMonth (LocalDateTime dateTime){
```

Comment:
If you are getting games at month, you expect a month to be passed (int or enum) and not a whole datetime.